salt/utils/network.py

```diff
@@ -589,7 +589,9 @@ def _interfaces_ifconfig(out):
                     if not salt.utils.is_sunos():
                         ipv6scope = mmask6.group(3) or mmask6.group(4)
                         addr_obj['scope'] = ipv6scope.lower() if ipv6scope is not None else ipv6scope
-                if addr_obj['address'] != '::' and addr_obj['prefixlen'] != 0:  # SunOS sometimes has ::/0 as inet6 addr when using addrconf
+                if salt.utils.is_sunos() and addr_obj['address'] != '::' and addr_obj['prefixlen'] != 0:  # SunOS sometimes has ::/0 as inet6 addr when using addrconf
+                    data['inet6'].append(addr_obj)
+                else:
                     data['inet6'].append(addr_obj)
```

Comment:
@terminalmage You are too right, trying to be to darn smart, tried to optimize the patch file when submitting and failed.  Will leave it as the patch in use.  Will fix.